### ISSUE TITLE
Support for more than Strings

### ISSUE THREAD

**Issue by cfoket** [MdFarhanRaja/SearchableSpinner#31]
Would it be possible to adapt the SpinnerDialog to support a list of Objects, rather than Strings? More specifically, I would like to see the type of the `item` parameter in `public void onClick(String item, int position)` be changed to `Object`.

I think PR #24 would be a good starting point. The first argument of `onSpinerItemClick.onClick(...)` could then just be `listView.getItemAtPosition(i)`.

The reason I am asking is because currently I have to resort to using two lists; one for the items I actually want to display, and a second one that contains their 'labels' (i.e. Strings), which I use when creating the spinner dialog. Then, I use the position I receive from the `onClick(String item,int position)` callback to determine which object (from the first list) was actually selected. However, this does not seem to work when the results of the spinner dialog are filtered. In that case I get the position in the filtered list, rather than the position in the unfiltered list, which is what I need. I think it would be much easier if the callback just returned the object that was selected.

**Comment by MdFarhanRaja** [comment]
Yes it's possible. For that i have to update my library and publish it.

I will update you once its done.

Thanks.

**Comment by cfoket** [comment]
Any updates on this yet? Thank you.

**Comment by neocodegh** [comment]
Any update on the library yet

**Comment by vamfixed** [comment]
Updates please?

**Comment by MdFarhanRaja** [comment]
Currently it's only supports string
